src/basic/process-util.c

```diff
@@ -87,8 +88,16 @@ int get_process_comm(pid_t pid, char **name) {
         r = read_one_line_file(p, name);
         if (r == -ENOENT)
                 return -ESRCH;
+        if (r < 0)
+                return r;
 
-        return r;
+        /* Strip unprintable characters, just in case */
+        for (a = b = *name; *a; a++)
+                if (isprint(*a))
+                        *(b++) = *a;
+
```

Comment:
Hm, maybe this should be remade on top of `cellescape`? At least when this gets logged in the journal, it seems wrong to just silently skip invalid characters.

Reply:
true, will rebase as soon as that's merged